udf_js/glean_percentile.sql

```diff
@@ -0,0 +1,51 @@
+CREATE OR REPLACE FUNCTION udf_js.glean_percentile(
+  percentile FLOAT64,
+  histogram ARRAY<STRUCT<key STRING, value FLOAT64>>,
+  type STRING
+)
+RETURNS FLOAT64
+LANGUAGE js
+AS
+  '''
+  if (percentile < 0 || percentile > 100) {
+      throw "percentile must be a value between 0 and 100";
+  }
+
+  let values = histogram.map(bucket => bucket.value);
+  let total = values.reduce((a, b) => a + b);
+  let normalized = values.map(value => value / total);
+
+  // Find the index into the cumulative distribution function that corresponds
+  // to the percentile. This undershoots the true value of the percentile.
+  let acc = 0;
+  let index = null;
+  for (let i = 0; i < normalized.length; i++) {
+      acc += normalized[i];
+      index = i;
+      if (acc >= percentile / 100) {
+          break;
+      }
+  }
+
+  // NOTE: we do not perform geometric or linear interpolation, but this would
+  // be the place to implement it.
+  return histogram[index].key;
```

Comment:
@mdboom I decided to rewrite the function to calculate percentiles without interpolation of the histogram values. I was wondering if you had any opinions on this? Note that this can be changed at a later time, since this is near the end of the transformation pipeline.

Reply:
I think this is fine.  One of the design decisions was to make sure the buckets are never so wide as to lose a lot of meaningful resolution.

However, I feel like this is fundamentally a DS question.  Corey Howgelund worked on the design of these distributions in the past.

Reply:
Thank you, I ended up filing a [Jira Issue [DS-430]](https://jira.mozilla.com/browse/DS-430) to figure out the best way to handle this.